core/src/main/java/com/linecorp/armeria/client/AbstractClientOptionsBuilder.java

```diff
@@ -60,7 +60,12 @@ protected AbstractClientOptionsBuilder(ClientOptions options) {
      */
     public AbstractClientOptionsBuilder options(ClientOptions options) {
         requireNonNull(options, "options");
-        options.asMap().values().forEach(this::option);
+        for (ClientOption<Object> option : options.options()) {
+            final Object value = options.getOrNull(option, false);
```

Comment:
This is not very efficient because we have to do a lookup for each option. How about making `AbstractOptions` implement `Iterable<AbstractOptionValue>`?

    public abstract class AbstractOptions<T extends AbstractOptionValue<?, ?>>
            implements Iterable<T>

Reply:
That sounds good. 👍